examples/Azure_IoT_Adu_ESP32/Azure_IoT_Adu_ESP32.ino

```diff
@@ -0,0 +1,1024 @@
+  }
+  else
+  {
+    Logger.Error(
+        "Error: " + String(rc) + " Received a message from an unknown topic: " + String(topic));
+  }
+}
+
+static esp_err_t mqtt_event_handler(esp_mqtt_event_handle_t event)
+{
+  switch (event->event_id)
+  {
+    int i, r;
+
+    case MQTT_EVENT_ERROR:
+      Logger.Info("MQTT event MQTT_EVENT_ERROR");
+      Logger.Info("ESP ERROR " + String(event->error_handle->esp_tls_last_esp_err));
+      Logger.Info("TLS ERROR " + String(event->error_handle->esp_tls_stack_err));
+      Logger.Info("ERROR TYPE " + String(event->error_handle->error_type));
+      Logger.Info(
+          "ESP Error " + String(esp_err_to_name(event->error_handle->esp_tls_last_esp_err)));
+      break;
+    case MQTT_EVENT_CONNECTED:
+      Logger.Info("MQTT event MQTT_EVENT_CONNECTED");
+
+      r = esp_mqtt_client_subscribe(mqtt_client, AZ_IOT_HUB_CLIENT_C2D_SUBSCRIBE_TOPIC, 1);
+      if (r == -1)
+      {
+        Logger.Error("Could not subscribe for cloud-to-device messages.");
+      }
+      else
+      {
+        Logger.Info("Subscribed for cloud-to-device messages; message id:" + String(r));
+      }
+
+      r = esp_mqtt_client_subscribe(
+          mqtt_client, AZ_IOT_HUB_CLIENT_PROPERTIES_MESSAGE_SUBSCRIBE_TOPIC, 1);
+      if (r == -1)
+      {
+        Logger.Error("Could not subscribe to properties messages.");
+      }
+      else
+      {
+        Logger.Info("Subscribed to properties messages; message id:" + String(r));
+      }
+
+      r = esp_mqtt_client_subscribe(
+          mqtt_client, AZ_IOT_HUB_CLIENT_PROPERTIES_WRITABLE_UPDATES_SUBSCRIBE_TOPIC, 1);
+      if (r == -1)
+      {
+        Logger.Error("Could not subscribe to writable property updates.");
+      }
+      else
+      {
+        Logger.Info("Subscribed to writable property updates; message id:" + String(r));
+      }
+
+      break;
+    case MQTT_EVENT_DISCONNECTED:
+      Logger.Info("MQTT event MQTT_EVENT_DISCONNECTED");
+      break;
+    case MQTT_EVENT_SUBSCRIBED:
+      Logger.Info("MQTT event MQTT_EVENT_SUBSCRIBED");
+      break;
+    case MQTT_EVENT_UNSUBSCRIBED:
+      Logger.Info("MQTT event MQTT_EVENT_UNSUBSCRIBED");
+      break;
+    case MQTT_EVENT_PUBLISHED:
+      Logger.Info("MQTT event MQTT_EVENT_PUBLISHED");
+      break;
+    case MQTT_EVENT_DATA:
+      Logger.Info("MQTT event MQTT_EVENT_DATA");
+
+      // Chunked data will not have a topic. Only copy when topic length is great
+      // than 0.
+      if (event->topic_len > 0)
+      {
+        for (i = 0; i < (SAMPLE_MQTT_TOPIC_LENGTH - 1) && i < event->topic_len; i++)
+        {
+          incoming_topic[i] = event->topic[i];
+        }
+        incoming_topic[i] = '\0';
+        Logger.Info("Topic: " + String(incoming_topic));
+      }
+
+      for (i = 0; i < (SAMPLE_MQTT_PAYLOAD_LENGTH - 1) && i < event->data_len; i++)
+      {
+        incoming_data[i] = event->data[i];
+      }
+      incoming_data[i] = '\0';
+      // Logger.Info("Data: " + String(incoming_data));
```

Comment:
Should this get uncommented to match the Topic?

Reply:
removing